rest/server_context.go

```diff
@@ -1093,6 +1093,11 @@ func dbcOptionsFromConfig(ctx context.Context, sc *ServerContext, config *DbConf
 		}
 	}
 
+	// In sync gateway version 4.0+ we do not support the disabling of use of xattrs
+	if !config.UseXattrs() {
+		return db.DatabaseContextOptions{}, fmt.Errorf("sync gateway requires shared_bucket_access to be enabled")
```

Comment:
This is an annoyingly named config option but I think this is a less confusing error message:

```suggestion
		return db.DatabaseContextOptions{}, fmt.Errorf("sync gateway requires enable_shared_bucket_access=true")
```